### PR TITLE
Correct the type of `offerId`

### LINKED ISSUE #302: Offer Operation: Exception shadow.com.google.gson.JsonSyntaxException: java.lang.NumberFormatException
### What version are you using?
Version `java-stellar-sdk 0.20.0`
I tested also on version 0.18.0 and 0.19.0 (the same result)

### What did you do?
Request for operations for block 32082240 with include_failed equals true
```
Page<OperationResponse> operationsPage = stellar.operations().forLedger(32082240).includeFailed(true).execute()
operationsPage.getNextPage(stellar.getHttpClient())
```

### What did you expect to see?
Response with success and failed operations

### What did you see instead?
Exception
```
Exception shadow.com.google.gson.JsonSyntaxException: java.lang.NumberFormatException: For input string: "137792008373673985"
```

###  Description
In classes:

- ManageBuyOfferOperationResponse
- ManageSellOfferOperationResponse
- CreatePassiveSellOfferOperationResponse

offerId has integer type but in **TradeEffectResponse** offerId has long type

In operation 
https://horizon.stellar.org/transactions/0decf364a116693f9d5a548cad501ccd0bed8b6b00a7d24ffc123c48d9c7acad/operations offer_id is long type. So in ManageBuyOfferOperationResponse is exception

```
shadow.com.google.gson.JsonSyntaxException: java.lang.NumberFormatException: For input string: "137792008373673985"
        at shadow.com.google.gson.internal.bind.TypeAdapters$7.read(TypeAdapters.java:228)
        at shadow.com.google.gson.internal.bind.TypeAdapters$7.read(TypeAdapters.java:218)
        at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
        at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
        at shadow.com.google.gson.Gson.fromJson(Gson.java:927)
        at shadow.com.google.gson.Gson.fromJson(Gson.java:994)
        at shadow.com.google.gson.Gson.fromJson(Gson.java:967)
        at org.stellar.sdk.responses.OperationDeserializer.deserialize(OperationDeserializer.java:57)
        at org.stellar.sdk.responses.OperationDeserializer.deserialize(OperationDeserializer.java:16)
        at shadow.com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69)
        at shadow.com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
        at shadow.com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82)
        at shadow.com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61)
        at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
        at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
        at shadow.com.google.gson.Gson.fromJson(Gson.java:927)
        at shadow.com.google.gson.Gson.fromJson(Gson.java:994)
        at org.stellar.sdk.responses.PageDeserializer.deserialize(PageDeserializer.java:47)
        at org.stellar.sdk.responses.PageDeserializer.deserialize(PageDeserializer.java:18)
        at shadow.com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69)
        at shadow.com.google.gson.Gson.fromJson(Gson.java:927)
        at shadow.com.google.gson.Gson.fromJson(Gson.java:892)
        at shadow.com.google.gson.Gson.fromJson(Gson.java:841)
        at org.stellar.sdk.requests.ResponseHandler.handleResponse(ResponseHandler.java:42)
        at org.stellar.sdk.responses.Page.getNextPage(Page.java:56)
```

Exception is when second line is executed
```
Page<OperationResponse> operationsPage = stellar.operations().forLedger(32082240).includeFailed(true).execute()
operationsPage.getNextPage(stellar.getHttpClient())

```

### PR DESCRIPTION
**Pull request by overcat**
Fixes #302

### CODE REVIEW

**Comment by overcat** [comment]
I don't know why the CI running result is not displayed, but all tests have passed locally, I believe this is an internal error of CircleCI.

File: CHANGELOG.md
Comment (leighmcculloch) [inline CHANGELOG.md:8]:
```suggestion
- Change the type of `offerId` from **java.lang.Integer** to **java.lang.Long** to align with the 64-bit size of offer IDs. ([#303](https://github.com/stellar/java-stellar-sdk/pull/303)).
```

**Review by leighmcculloch — APPROVED** [review]
This looks like the correct fix to me. IDs in other response types use `Long` rather than `Integer` and I'm guessing we just missed this case whenever the IDs were expanded to be 64 bit.

I'm not an expert on this change though and so I don't think this should be merged until someone from the @stellar/horizon-committers can confirm that we're not missing something here.

**Review by bartekn — APPROVED** [review]
LGTM, looks like we missed it when upgrading to Protocol 11.

**Pull request merged**
